### PR TITLE
[JENKINS-73133] Adapt GitHub for Jetty 12 (EE 8)

### PR DESCRIPTION
**Pull request by basil**
See [JENKINS-73133](https://issues.jenkins.io/browse/JENKINS-73133). Starting on Tuesday, August 13, the 2.472 weekly release will contain Jetty 12 EE 8. For this plugin to continue to be testable in PCT (Plugin Compatibility Tester) with the upcoming 2.472 weekly release, we must adapt its test suite to work on Jetty 12 EE 8, including merging and releasing this PR.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -54,6 +54,9 @@
         <release.skipTests>false</release.skipTests>
         <tagNameFormat>v@{project.version}</tagNameFormat>
         <useBeta>true</useBeta> <!-- For Jenkins.MANAGE permission -->
+        <!-- TODO JENKINS-73339 until in parent POM -->
+        <jenkins-test-harness.version>2250.v03a_1295b_0a_30</jenkins-test-harness.version>
+        <maven.compiler.testRelease>17</maven.compiler.testRelease>
```
Comment (basil) [inline pom.xml:59]: Temporary change to run tests with Jetty 12 (which requires Java 17). Note that PCT (Plugin Compatibility Tester) automatically applies these changes when this plugin is tested in `jenkinsci/bom` against a Jenkins core that contains Jetty 12 (e.g., the 2.472 pre-release, or the 2.472 weekly release that is scheduled for next Tuesday, August 13).

File: src/test/java/org/jenkinsci/plugins/github/config/GitHubServerConfigIntegrationTest.java [hunk 2]
```diff
@@ -14,10 +14,9 @@
 import net.sf.json.JSONObject;
 import org.eclipse.jetty.server.Server;
 import org.eclipse.jetty.server.ServerConnector;
-import org.eclipse.jetty.servlet.DefaultServlet;
-import org.eclipse.jetty.servlet.ServletContextHandler;
-import org.eclipse.jetty.servlet.ServletContextHandler.Context;
-import org.eclipse.jetty.servlet.ServletHolder;
+import org.eclipse.jetty.ee8.servlet.DefaultServlet;
+import org.eclipse.jetty.ee8.servlet.ServletContextHandler;
+import org.eclipse.jetty.ee8.servlet.ServletHolder;
```
Comment (basil) [inline src/test/java/org/jenkinsci/plugins/github/config/GitHubServerConfigIntegrationTest.java:19]: Jetty 11 and older were tied to a particular Servlet API version, but Jetty 12 and newer features a modular architecture that allow Jetty to be used with any Servlet API version. For EE 8, the classes have now been moved to these packages.

This is needed for this plugin's test suite to work against Jetty 12 EE 8, which PCT will start running after the 2.472 weekly release on Tuesday, August 13.

**Comment by basil** [comment]
The [CI build is green](https://ci.jenkins.io/job/Plugins/job/github-plugin/job/PR-380/6/).

@KostyaSha Can this PR please be released before (or shortly after) the Jenkins 2.472 weekly release, currently scheduled for Tuesday, August 13?

**Comment by basil** [comment]
Thank you very much!

**Comment by KostyaSha** [comment]
you are welcome

**Pull request merged**